### PR TITLE
Added origins:disable_regen power type

### PR DESCRIPTION
**Pull request by xMGZx**
An idea for a power which disables regeneration on a player-by-player basis (instead of through /gamerule naturalRegeneration false) put into code. You can add this code in manually if you don't want to accept this pull request.

### CODE REVIEW

**Comment by apace100** [comment]
Could you modify this commit so it merges into the feature branch? :)

File: src/main/java/io/github/apace100/origins/power/PowerType.java [hunk 1]
```diff
@@ -35,6 +35,9 @@ public PowerType setHidden() {
         return this;
     }
 
+    public void onRemoved() {}
+    public void onLost() {}
```
Comment (apace100) [inline src/main/java/io/github/apace100/origins/power/PowerType.java:39]: Why is this here? :D

File: src/main/resources/data/origins/powers/disable_regen.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+    "type": "origins:disable_regen",
+    "name": "No Regen",
+    "description": "You cannot heal through normal means like food."
+}
```
Comment (apace100) [inline src/main/resources/data/origins/powers/disable_regen.json:5]: There shouldn't be a default power for this if it's not used.

File: src/main/java/io/github/apace100/origins/power/PowerType.java (hunk 1 above)
Comment (xMGZx) [reply src/main/java/io/github/apace100/origins/power/PowerType.java]: For some reason fabric was giving me an error telling me those default methods were required there and no other branch complained about it ever again.
